runtime/src/lib.rs

```diff
@@ -71,6 +71,9 @@ pub enum Error {
 /// running tasks.
 pub trait Runner {
     /// Start running a root task.
+    ///
+    /// TODO: It is up to a runtime implementation to handle whether a finished root task should
```

Comment:
Discovered inconsistent behavior between `runtime::deterministic` and `runtime::tokio` around task execution when `start` returns. We should consider enforcing similar behavior but at the least should document.

Reply:
Probably best to kill everything when start returns.